phd-tests/framework/src/test_vm/config.rs

```diff
@@ -257,10 +257,14 @@ impl<'dr> VmConfig<'dr> {
         // in the correct order: boot disk first, then in the data disks'
         // iteration order.
         let all_disks = self.disks.iter().zip(disk_handles.iter());
-        for (idx, (req, hdl)) in all_disks.enumerate() {
-            let device_name = format!("disk-device{}", idx);
+        for (req, hdl) in all_disks {
             let pci_path = PciPath::new(0, req.pci_device_num, 0).unwrap();
-            let (backend_name, backend_spec) = hdl.backend_spec();
+            let (device_name, backend_spec) = hdl.backend_spec();
+            // propolis-server uses a disk's name from the API as its device
+            // name. It then derives a backend name from the disk name. Match
+            // that name derivation here so PHD tests match the API as used by
+            // Nexus.
+            let backend_name = format!("{}-backend", device_name);
```

Comment:
This also requires an update to `VmSpec::refresh_crucible_backends` so that it can find the correct backend name for each disk handle in the resulting spec. (This is the cause of the most recent commits' CI failures--the Crucible migration tests are failing because the framework isn't correctly bumping the client generation numbers on the disks, which is happening because the backend elements in the instance spec don't have the names `refresh_crucible_backends` expects.)

I would probably plumb all this a bit differently so that disk objects return the backend names we expect:

- have separate `DiskDeviceName` and `DiskBackendName` newtypes, with an associated function on the former that generates the latter
- make `DiskFactory`'s functions take `DiskDeviceName`s instead of bare `String`s
- make `DiskConfig::backend_spec` return a `DiskBackendName`
- do whatever plumbing is needed in DiskFactory or its callees to do the correct device -> backend name conversion

Reply:
> cause of the most recent commits' CI failures

ah, i trying to figure out caused that outcome. ok, that all makes some sense

Reply:
rereading this, i see it's what you meant the entire time, but i ended up at https://github.com/oxidecomputer/propolis/pull/756/commits/3a3e753ca28a383f48eea33673f339f835a98546 - i'd like to just have whatever relation between `DeviceName`/`BackendName` be _the_ function between them instead of simply matching what `parse_disk_from_request` does, but i think that's much more invasive than is appropriate here..